controllers/component_build_controller_test.go

```diff
@@ -136,7 +137,72 @@ var _ = Describe("Component initial build controller", func() {
 				},
 			}
 			Expect(k8sClient.Create(ctx, component)).Should(Succeed())
+		})
+		AfterEach(func() {
+			// Clean up
+			deleteComponentInitialPipelineRuns(resourceKey)
+			deleteComponent(resourceKey)
+			deleteConfigMap(prepare.BuildBundleConfigMapName, HASAppNamespace)
+			Expect(k8sClient.Delete(ctx, gitSecret)).Should(Succeed())
+		})
+
+		It("should create build objects when secret missing", func() {
+
+			setComponentDevfileModel(resourceKey)
+
+			// Wait until all resources created
+			ensureOneInitialPipelineRunCreated(resourceKey)
+
+			// Check that git credentials secret is annotated
+			Expect(k8sClient.Get(ctx, types.NamespacedName{Name: GitSecretName, Namespace: HASAppNamespace}, gitSecret)).Should(Succeed())
+			tektonGitAnnotation := gitSecret.ObjectMeta.Annotations["tekton.dev/git-0"]
+			Expect(tektonGitAnnotation).To(Equal("https://github.com"))
+
+			// Check that the pipeline service account has been linked with the Github authentication credentials
+			var pipelineSA corev1.ServiceAccount
+			Expect(k8sClient.Get(ctx, types.NamespacedName{Name: "pipeline", Namespace: HASAppNamespace}, &pipelineSA)).Should(Succeed())
+
+			secretFound := false
+			for _, secret := range pipelineSA.Secrets {
+				if secret.Name == GitSecretName {
+					secretFound = true
+					break
+				}
+			}
+			Expect(secretFound).To(BeTrue())
 
+			// Check the pipeline run and its resources
+			pipelineRuns := listComponentInitialPipelineRuns(resourceKey)
+			Expect(len(pipelineRuns.Items)).To(Equal(1))
+			pipelineRun := pipelineRuns.Items[0]
+
+			Expect(pipelineRun.Spec.Params).ToNot(BeEmpty())
+			for _, p := range pipelineRun.Spec.Params {
+				if p.Name == "output-image" {
+					Expect(p.Value.StringVal).To(Equal("docker.io/foo/customized:default-test-component"))
+				}
+				if p.Name == "git-url" {
+					Expect(p.Value.StringVal).To(Equal(SampleRepoLink))
+				}
+			}
+
+			Expect(pipelineRun.Spec.PipelineRef.Bundle).To(Equal(buildBundle))
+
+			Expect(pipelineRun.Spec.Workspaces).To(Not(BeEmpty()))
```

Comment:
We have duplicated most of the test case. Does it makes sense to have a separate function within the test context that tests common stuff?